app/processors/transform_processor.py

```diff
@@ -4,17 +4,26 @@
 from structlog import get_logger
 
 
-class Processor:
-    """Base class calls transformer and passes results to ftp"""
+class TransformProcessor:
+    """Transforms and passes results to ftp"""
+
+    def __init__(self, survey, ftp_conn):
+        """
+           survey: survey document
+           ftp_conn: ftp connection , passed in for testing purposes
```

Comment:
Yes and no. It is a needed component. However it is passed in rather than being having a local initialiser so that we can pass in a dummy one during testing and hence remove that dependency.